src/auth/src/credentials/service_account.rs

```diff
@@ -32,33 +94,117 @@ use time::OffsetDateTime;
 const DEFAULT_SCOPES: &str = "https://www.googleapis.com/auth/cloud-platform";
 
 pub(crate) fn creds_from(js: serde_json::Value) -> Result<Credential> {
-    let service_account_info =
-        serde_json::from_value::<ServiceAccountInfo>(js).map_err(CredentialError::non_retryable)?;
-    let token_provider = ServiceAccountTokenProvider {
-        service_account_info,
-    };
-    let token_provider = TokenCache::new(token_provider);
-
-    Ok(Credential {
-        inner: Arc::new(ServiceAccountCredential { token_provider }),
-    })
+    let service_account_key =
+        serde_json::from_value::<ServiceAccountKey>(js).map_err(CredentialError::non_retryable)?;
+    Ok(Builder::default()
+        .service_account_key(service_account_key)
+        .build())
+}
+
+#[derive(Default)]
+pub struct Builder {
+    service_account_key: ServiceAccountKey,
+    aud: Option<String>,
+    scopes: Option<String>,
+    quota_project_id: Option<String>,
+}
+
+impl Builder {
+    /// Sets the [service account key] in [aip/4112] format for this credential.
+    ///
+    /// [aip/4112]: https://google.aip.dev/auth/4112
+    /// [service account key]: https://cloud.google.com/iam/docs/keys-create-delete#creating
+    pub fn service_account_key(mut self, service_account_key: ServiceAccountKey) -> Self {
+        self.service_account_key = service_account_key;
+        self
+    }
+
+    /// Sets the audience for this credential.
+    ///
+    /// `aud` is a [JWT] claim specifying intended recipient(s) of the token,
+    /// that is, a service(s) or resource(s).
+    /// This cannot be used at the same time as the scopes claim.
+    /// The value should be https://{SERVICE}/. (e.g. `https://pubsub.googleapis.com/`)
+    ///
+    /// [JWT]: https://google.aip.dev/auth/4111
+    pub fn aud<S: Into<String>>(mut self, aud: S) -> Self {
+        self.aud = Some(aud.into());
+        self
+    }
+
+    /// Sets the [scopes] for this credentials.
+    ///
+    /// `scopes` is a [JWT] claim specifying requested permission(s) for the token.
+    /// This cannot be used at the same time as the aud claim.
+    /// Multiple scopes can be specified using a single space (" ") as delimiter.
+    ///
+    /// [JWT]: https://google.aip.dev/auth/4111
+    /// [scopes]: https://developers.google.com/identity/protocols/oauth2/scopes
+    pub fn scopes<S: Into<String>>(mut self, scopes: S) -> Self {
+        self.scopes = Some(scopes.into());
+        self
+    }
+
+    /// Sets the [quota project] for this credential.
+    ///
+    /// In some services, you can use a service account in
+    /// one project for authentication and authorization, and charge
+    /// the usage to a different project. This may require that the
+    /// service account has `serviceusage.services.use` permissions on the quota project.
+    ///
+    /// [quota project]: https://cloud.google.com/docs/quotas/quota-project
+    pub fn quota_project_id<S: Into<String>>(mut self, quota_project_id: S) -> Self {
```

Comment:
I should have noticed this earlier.  All the other builders use `with_` or `set_` . It is a bit sad that auth uses a different convention.

Reply:
It is not too late to change. Other than this one, we currently have only one other builder I believe (and that may not even be released yet).